aexpect/ops_windows.py

```diff
@@ -0,0 +1,578 @@
+    _run_as(cmd)
+    LOG.debug("Waiting %s seconds for the command to finish", timeout)
+    # no need to capture anything, we redirected the output
+    session.read_up_to_prompt(timeout=timeout)
+
+    # after reading the output from the file we can remove it
+    output = cat(session, outfile).strip()
+    remove(session, outfile)
+    return output
+
+
+def kill_program(session, program, kill_children=True):
+    """
+    Forcefully kill a program.
+
+    :param session: session of the Windows guest
+    :type session: :py:class:`aexpect.client.RemoteSession`
+    :param str program: name of the program to kill
+    :param bool kill_children: whether to kill child processes
+    """
+    cmd = f"taskkill /f /im \"{program}\""
+    if kill_children:
+        cmd += " /t"
+    session.cmd(cmd)
+
+
+def wait_process_end(session, process, timeout=30):
+    """
+    Wait until a process has fully ended using PowerShell.
+
+    :param session: session of the Windows guest
+    :type session: :py:class:`aexpect.client.RemoteSession`
+    :param str process: name of the process to track
+    :param int timeout: timeout to wait before giving up
+    :raises: :py:class:`TimeoutError` if an error occurred
+    :raises: :py:class:`RuntimeError` if an error occurred
+    """
+    LOG.info("Waiting for process %s to end using PowerShell", process)
+
+    # give some extra timeout to wait for PowerShell to return
+    ps_cmd(session, f"Wait-Process -Name {process} -Timeout {timeout}",
+            timeout=timeout+5)
+
+
+def kill_session(session):
+    """
+    Kill the rss process responsible for a session.
+
+    :param session: session of the Windows guest
+    :type session: :py:class:`aexpect.client.RemoteSession`
+    """
+    LOG.info("Killing session for the user %s", session.username)
+    cmd = "(Get-WmiObject Win32_Process -Filter ProcessId=$PID).ParentProcessId"
+    cmd_pid = ps_cmd(session, cmd)
+    LOG.debug("Got cmd.exe with PID %s", cmd_pid)
+
+    rss = f"(Get-WmiObject Win32_Process -Filter ProcessId={cmd_pid}).ParentProcessId"
+    rss_pid = ps_cmd(session, rss)
+    LOG.debug("Got rss.exe with PID %s", rss_pid)
+
+    session.sendline(f"taskkill /F /PID {rss_pid}")
+    # read everything left to read
+    session.read_nonblocking(internal_timeout=0, timeout=3)
+    # make sure we got the right session (session.is_responsive() does not always work here)
+    try:
+        session.cmd("whoami", timeout=5)
+    except ShellProcessTerminatedError:
+        # session ended correctly
+        pass
+    else:
+        raise ProcessLookupError(f"Could not kill rss.exe for user {session.username}")
+
+    session.close()
+    LOG.info("Session successfully killed")
+
+
+###############################################################################
+# network ops
+###############################################################################
+
+
+def find_unused_port(session, start_port=10024):
```

Comment:
Is this a typo or are ports 10024+ special in windows?

Reply:
The main reason is that typical ports used by Avocado VT for windows vms are:
```
        shell_client = nc
        shell_port = 10022
        file_transfer_client = rss
        file_transfer_port = 10023
```
but I guess we could also set the default to something more standard outside of this scope. Do you have any specific suggestions?